SDAPackage/SDA/obstacle_map.py

```diff
@@ -27,11 +27,56 @@ def add_obstacle(self, obstacle_to_add):
         :param obstacle_to_add: The obstacle to add to the map
         :type obstacle_to_add: StationaryObstacle
         """
+        final_obstacles = np.array([])
         if self.obstacles.size != 0:
-            self.obstacles = np.hstack((self.obstacles, obstacle_to_add))
+            final_obstacles = np.hstack((self.obstacles, obstacle_to_add))
+            final_obstacles = overlapping_obstacles(final_obstacles)
+            self.obstacles = np.hstack((self.obstacles, final_obstacles))
```

Comment:
Why are you using ```numpy.hstack``` here? You should just be setting ```self.obstacles = self.overlapping_obstacles()```

Reply:
done